tc-client/src/main/java/com/tc/object/EntityClientEndpointImpl.java

```diff
@@ -220,9 +226,12 @@ public void didCloseUnexpectedly() {
     }
   }
 
-  private void checkEndpointOpen() {
+  private synchronized void checkEndpointOpen(boolean close) {
```

Comment:
Would seem simpler to keep `checkEndpointOpen()` as-is and have a new synchronized method for the close call?